framework/include/actioncomponents/CylinderComponent.h

```diff
@@ -0,0 +1,39 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#pragma once
+
+// MOOSE includes
+#include "ActionComponent.h"
+#include "PhysicsComponentHelper.h"
+
+/**
+ * Cylinder on which one can define a Physics. The mesh is automatically created
+ */
+class CylinderComponent : public virtual ActionComponent, public PhysicsComponentHelper
```

Comment:
No purpose here. But when the next helper comes along that also derives from ActionComponent, and one wants to use two helper that's when the virtual inheritance comes in handy

Reply:
well no purpose isnt exactly true. Imagine I split here between components that use a Physics (parameter) and components that dont (only geometry, or define their own kernels)

then we decide to split between 1D 2D and 3D components for base classes. Then we end up having to have 3 derived classes for PhyisicsComponent and the others

Reply:
So what's the breakdown of the virtual inheritance vs. interface approach? I think if you do go the virtual inheritance route, I feel like the `Helper` in `PhysicsComponentHelper` doesn't belong, since children are Physics components, not just helpers.

Reply:
I can rename to PhysicsComponentBase.
I m not sure which approach is superior at this point